.github/workflows/build-macos.yml

```diff
@@ -0,0 +1,122 @@
+name: Build and test on macos
+
+on:
+  pull_request:
+    types:
+      - opened             # triggers build when opened
+      - closed             # triggers build when merged
+      - synchronize        # triggers build when commits are pushed to HEAD
+      - auto_merge_enabled # triggers build when auto-merge is enabled
+    branches: 
+      - master
+      - 'release/v[0-9].[0-9].[0-9]'
+      - 'feature/**'
+  # Allow triggering this workflow manually from the actions tab
+  workflow_dispatch:
+
+jobs:
+  
+  build:
+    
+    # Build strategy
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          #- 'ubuntu-latest'
+          - 'macos-latest'
+        build_type:
+          - 'Release'
+          #- 'Debug'
+          #- 'DebugWithRelInfo '
+
+    # Build platform
+    runs-on: ${{ matrix.platform }}
+
+    name: ${{ matrix.platform }}-${{ matrix.build_type }}
+
+    # default compiler on macos is clang, switch to gcc 11. Specifying the version is necessary.
+    # It seems like gcc and g++ are symbolic links to the default clang and clang++ compilers, respectively.
+    # CMAKE_CXX_COMPILER_ID will evaluate to AppleClang rather than GNU on macos.
+    env:
+      CC: gcc-11
+      CXX: g++-11
+    
+      # Build steps
+    steps:
+
+      # Step: Checkout
+      - name: Checkout
+        uses: actions/checkout@v3
+        # Workaround for getting "git describe --tags" to work in cmake/get_version_from_git.cmake (Build step)
+        with:
+          fetch-depth: 0
+
+      # Step: Set up Python
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+
+      # Step: Install Python dependencies
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install wheel
```

Comment:
python -m pip install wheel numpy matplotlib pytest. you can use one line

Reply:
Of course. It is nicer to have separate commands if errors happen.